### PR TITLE
Serve module source requests from index

### PR DESCRIPTION
**Pull request by habdelra**
This PR allows module source files to be served from the index instead of the filesystem

### CODE REVIEW

File: packages/runtime-common/indexer.ts [hunk 1]
```diff
@@ -256,9 +276,10 @@ export class Indexer {
       isolatedHtml,
       searchDoc,
       types,
-      indexedAt,
+      indexedAt: indexedAt != null ? parseInt(indexedAt) : null,
       source,
       deps,
+      lastModified: parseInt(lastModified),
```
Comment (backspace) [inline packages/runtime-common/indexer.ts:282]: if these are big integers do we have to think about overflow?

**Comment by backspace** [comment]
non-review question: I see the `X-Boxel-Use-WIP-Index` header is checked to enable this, I thought I was exercising it in #1242 but probably not now that I’ve noticed this, is there an easy way to globally enable it in that branch?

**Review by backspace — APPROVED** [review]

File: packages/runtime-common/indexer.ts (hunk 1 above)
Comment (habdelra) [reply packages/runtime-common/indexer.ts:282]: This is specifically to store Date.now(). I don’t think overflow will be an issue.

**Comment by habdelra** [comment]
> non-review question: I see the `X-Boxel-Use-WIP-Index` header is checked to enable this, I thought I was exercising it in #1242 but probably not now that I’ve noticed this, is there an easy way to globally enable it in that branch?

I don’t understand the question. We have unit tests that specifically exercise this header. This header is a particular mechanism used in order to generate the index. I’m not sure what you mean by “globally enable it”

**Comment by backspace** [comment]
> I don’t understand the question. We have unit tests that specifically exercise this header. This header is a particular mechanism used in order to generate the index. I’m not sure what you mean by “globally enable it”

I merged this branch into #1242 to run in Actions, does this mean the WIP index is used in most of the tests or would I need to somehow opt in? I’m still hoping this magically makes Firefox faster in CI haha

**Comment by habdelra** [comment]
> > I don’t understand the question. We have unit tests that specifically exercise this header. This header is a particular mechanism used in order to generate the index. I’m not sure what you mean by “globally enable it”
> 
> I merged this branch into #1242 to run in Actions, does this mean the WIP index is used in most of the tests or would I need to somehow opt in? I’m still hoping this magically makes Firefox faster in CI haha

Oh so I think there is a big misunderstanding. There are always 2 versions of the index, a production index and a “WIP” index. The WIP index is the index whose items include index entries that are higher than the current realm version. The idea is that we have indexing that occurs in batches, while the batch is processing there is a special “work in progress” index that is seen by the indexer only. For normal requests they will see the production index, meaning that the changes in the batch are not yet live. Only when the batch is complete then the WIP index is promoted to the production index. This way the indexing is consistent. Clients will not see partial updates—like if there was a module that lots of instances depend on that introduced a new computed, the update module and all the dependent instances are updated atomically. The client will never see some instances of the card that have a new computed and some that don’t.

**Review by jurgenwerk — APPROVED** [review]

**Comment by jurgenwerk** [comment]
if I may add a small comment to the above confusion - I also tend to automatically associate "WIP" nomenclature with something that is yet to be included/implemented fully (from the application point of view). Now that I got used to the fact that we use that name for the "in progress index" it's really not a big deal (also it's really well documented) but maybe there's a chance it could be confusing for others contributors in the future too

**Comment by backspace** [comment]
> Oh so I think there is a big misunderstanding.

correct, thanks for the explanation 😆

**Pull request merged**